api/src/main/java/org/openmrs/module/idgen/validator/AutoGenerationOptionValidator.java

```diff
@@ -0,0 +1,27 @@
+package org.openmrs.module.idgen.validator;
+
+import org.openmrs.module.idgen.AutoGenerationOption;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+public class AutoGenerationOptionValidator implements Validator {
+
+	@Override
+	public boolean supports(Class<?> clazz) {
+
+		return AutoGenerationOptionValidator.class.isAssignableFrom(clazz);
+	}
+
+	@Override
+	public void validate(Object object, Errors errors) {
+		AutoGenerationOption autoGenaratorionOption = (AutoGenerationOption) object;
+
+		
+		if(!autoGenaratorionOption.isManualEntryEnabled() && autoGenaratorionOption.isAutomaticGenerationEnabled()) {
+			errors.reject("Manual Entry should be set to true");
```

Comment:
The case that shouldn't be allowed is when _manuel entry_ and _automatic generation_ **are both disabled.**